metadata-ingestion/src/datahub/ingestion/api/source.py

```diff
@@ -205,6 +206,10 @@ class SourceReport(Report):
         default_factory=lambda: defaultdict(lambda: defaultdict(LossyList))
     )
 
+    subtypes: Dict[str, Dict[str, int]] = field(
+        default_factory=lambda: defaultdict(lambda: defaultdict(int))
+    )
```

Comment:
imo the entities dict above should be broken down by subtype. having a separate subtypes list is marginally useful, but seeing e.g. upstreamLineage coverage broken down between Tables and Views would be super useful

This change is somewhat useful but it could be significantly more impactful